core/emulate.c

```diff
@@ -1069,20 +1106,27 @@ em_status_t EMCALL em_emulate_insn(struct em_context_t *ctxt)
     }
 
     // Input operands
-    if (!(opcode->flags & INSN_NOFLAGS)) {
-        ctxt->rflags = ctxt->ops->read_rflags(ctxt->vcpu);
-    }
-    if (!(opcode->flags & INSN_MOV)) {
-        rc = operand_read(ctxt, &ctxt->dst);
+    if (opcode->flags & INSN_BITOP &&
+        ctxt->dst.type == OP_MEM && ctxt->src1.type == OP_REG) {
+        rc = operand_read_bitop(ctxt);
+        if (rc != EM_CONTINUE)
+            goto exit;
+    } else {
+        if (!(opcode->flags & INSN_NOFLAGS)) {
```

Comment:
Why move this condition here? Any good reason to skip `read_rflags` for the `BT* mN, rN` case?

Reply:
Sorry, this was a typo, I've moved it out of the `else` case in 9a3e342 (and placed it before the `if`).